src/NUnitFramework/framework/Api/DefaultTestAssemblyBuilder.cs

```diff
@@ -258,7 +258,12 @@ private TestSuite BuildTestAssembly(Assembly assembly, string assemblyNameOrPath
 
             testAssembly.ApplyAttributesToTest(assembly);
 
-            testAssembly.Properties.Set(PropertyNames.ProcessId, System.Diagnostics.Process.GetCurrentProcess().Id);
+            try
+            {
+                testAssembly.Properties.Set(PropertyNames.ProcessId, System.Diagnostics.Process.GetCurrentProcess().Id);
```

Comment:
Also starting in .NET 5, we should use the much less wasteful `Environment.ProcessId`. Do you know if Blazor still implements `Environment.ProcessId` to throw the same exception?

Reply:
I guess we could also call https://docs.microsoft.com/en-us/dotnet/api/system.operatingsystem.isbrowser?view=net-5.0 inside the `#if NET5_0` section even if Environment.ProcessId does throw, rather than catching an exception.

Reply:
I think I would prefer that solution, as I think we should avoid catching exceptions with empty error handlers if it is possible.